### PR TITLE
Add test to illustrate how to call actions locally

### PR DESCRIPTION
**Pull request by PierreFritsch**
Add the code sample from the capire section [Unbound Actions / Functions](https://cap.cloud.sap/docs/node.js/services#-unbound-actions--functions) as a test in the code sample repository.

### CODE REVIEW

**Comment by PierreFritsch** [comment]
I'd need help on this one - my new test fails with:

> TypeError: Cannot read property 'SELECT' of undefined

**Comment by PierreFritsch** [comment]
The test file [consuming-services.test.js](https://github.com/SAP-samples/cloud-cap-samples/blob/master/test/consuming-services.test.js) might be a good place to add this code sample, but it seems to be set up to serve only the `AdminService` which, in contrast with the `CatalogService`, doesn't have an action that can be called.

**Comment by chgeo** [comment]
Full error is 
```
    TypeError: Cannot read property 'SELECT' of undefined

       6 |   }
       7 |
    >  8 |   if (query.SELECT) {
         |             ^
       9 |     return read(model, dbc, query, req)
      10 |   }
      11 |

      at SQLiteDatabase._run (../../cds/libx/_runtime/db/query/run.js:8:13)
      at SQLiteDatabase.call (../../cds/libx/_runtime/sqlite/Service.js:74:19)
      at next (../../cds/lib/serve/Service-dispatch.js:74:36)
      at SQLiteDatabase.handle (../../cds/lib/serve/Service-dispatch.js:72:16)
```

**Comment by chgeo** [comment]
Internal ticket is cap/issues#9798

**Comment by PierreFritsch** [comment]
Updated the code sample to reflect O.T.'s internal comment on October 4. What did I miss?

```
FAIL test/consuming-actions.test.js
  ● Consuming actions locally › calls unbound actions
      21 |
      22 |     // basic variant using srv.send
    > 23 |     const res1 = await cats.send("submitOrder", {
         |                             ^
      24 |       book: 111,
      25 |       quantity: 1,
      26 |     });

      at Request.reject (node_modules/@sap/cds/lib/req/request.js:65:39)
      at _reject (node_modules/@sap/cds/libx/_runtime/common/generic/auth.js:37:16)
      at CatalogService.handler (node_modules/@sap/cds/libx/_runtime/common/generic/auth.js:638:57)
      at CatalogService.handle (node_modules/@sap/cds/lib/serve/Service-dispatch.js:64:53)
      at CatalogService.handle (node_modules/@sap/cds/libx/_runtime/cds-services/services/Service.js:147:18)
      at CatalogService.dispatch (node_modules/@sap/cds/lib/serve/Service-dispatch.js:47:15)
      at CatalogService.dispatch (node_modules/@sap/cds/lib/serve/Service-dispatch.js:20:17)
      at CatalogService.send (node_modules/@sap/cds/lib/serve/Service-api.js:38:17)
      at Object.<anonymous> (test/consuming-actions.test.js:23:29)
```

**Comment by PierreFritsch** [comment]
> looks like the action was rejected due to missing authorization.
> 
> @requires: 'authenticated-user'

(Thank you S.!)

**Comment by PierreFritsch** [comment]
Any further details on how to:

> use a [manual transaction](https://cap.cloud.sap/docs/node.js/transactions#manual-transactions) to create a continuation with an [authenticated-user](https://cap.cloud.sap/docs/guides/authorization#pseudo-roles).

would be highly welcome...

(I couldn't figure out how to assemble the pieces mentioned in the documentation yet.)

**Review by danjoa — APPROVED** [review]
👍 Looks pretty good. Thanks Pierre!
Remark: another valid way to bypass access control in tests is this:
```js
cds.User.default = cds.User.Privileged
```
e.g. also [done here](https://github.com/SAP-samples/cloud-cap-samples/blob/1fde0c928be8804ef678bedde9320859f4c915f8/test/custom-handlers.test.js#L3)

**Comment by gregorwolf** [comment]
I like the approach providing the user explicitly as this will actually test the authorizations.

**Comment by chgeo** [comment]
@sjvans since this quite and old change, could you review this and check for proper usage of APIs (`tx` etc)?

File: test/consuming-actions.test.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+const cds = require("@sap/cds/lib");
```
Comment (sjvans) [inline test/consuming-actions.test.js:1]:
```suggestion
const cds = require("@sap/cds");
```

File: test/consuming-actions.test.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    // Read the stock before the action is called
+    stockBefore = (await cats.get(Books, BOOK_ID)).stock;
+
+    // Use a manual transaction to create a continuation with an authenticated user
+    tx = await cats.tx({user: "alice"});
+  });
+
+  afterEach(async() => {
+    await tx.rollback();
+  })
```
Comment (sjvans) [inline test/consuming-actions.test.js:35]:
```suggestion
```

File: test/consuming-actions.test.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  it("calls unbound actions - basic variant using srv.send", async () => {
+    const res1 = await tx.send("submitOrder", {
+      book: BOOK_ID,
+      quantity: QUANTITY,
+    });
```
Comment (sjvans) [inline test/consuming-actions.test.js:41]:
```suggestion
    // Use a managed transaction to create a continuation with an authenticated user
    const res1 = await cats.tx({ user: "alice" }, () => {
      return cats.send("submitOrder", { book: BOOK_ID, quantity: QUANTITY });
    });
```

File: test/consuming-actions.test.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    expect(res1.stock).to.eql(stockBefore - QUANTITY);
+  });
+
+  it("calls unbound actions - named args variant", async () => {
+    const res2 = await tx.submitOrder({ book: BOOK_ID, quantity: QUANTITY });
```
Comment (sjvans) [inline test/consuming-actions.test.js:46]:
```suggestion
    // Use a managed transaction to create a continuation with an authenticated user
    const res2 = await cats.tx({ user: "alice" }, () => {
      return cats.submitOrder({ book: BOOK_ID, quantity: QUANTITY });
    });
```

File: test/consuming-actions.test.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    expect(res2.stock).to.eql(stockBefore - QUANTITY);
+  });
+
+  it("calls unbound actions - positional args variant", async () => {
+    const res3 = await tx.submitOrder(BOOK_ID, QUANTITY);
```
Comment (sjvans) [inline test/consuming-actions.test.js:51]:
```suggestion
    // Use a managed transaction to create a continuation with an authenticated user
    const res3 = await cats.tx({ user: "alice" }, () => {
      return cats.submitOrder(BOOK_ID, QUANTITY);
    });
```

**Comment by sjvans** [comment]
> @sjvans since this quite and old change, could you review this and check for proper usage of APIs (`tx` etc)?

should use managed tx, see my suggestions

File: test/consuming-actions.test.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+const { expect } = cds.test(
+  "serve",
+  "CatalogService",
+  "--from",
+  "@capire/bookshop,@capire/common",
+  "--in-memory"
+);
+
+describe("Consuming actions locally", () => {
+  let cats, CatalogService, Books, tx, stockBefore;
```
Comment (sjvans) [inline test/consuming-actions.test.js:11]:
```suggestion
  let cats, CatalogService, Books, stockBefore;
```

**Review by sjvans — APPROVED** [review]

**Pull request merged**